app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js

```diff
@@ -218,6 +218,28 @@ window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {};
             domainsArrays.push(domainWithoutWww)
           }
         }
+      },
+
+      // The index and index_total properties are usually set by the application's templating engine however in instances where this isn't possible,
+      // we use JavaScript to calculate the index_total and append it to data-ga4-link and add an index property to each link
+      setIndexes: function (module) {
+        var linksToBeIndexed = []
+
+        // Only index links that are not search results
+        // querySelectorAll(a:not['data-ga4-ecommerce-path']) would have been used if not for lack of support in older browsers
```

Comment:
Not sure this comment is needed? I'd put it in the PR description if you want to document it.